src/ae.c

```diff
@@ -193,6 +195,9 @@ void aeDeleteFileEvent(aeEventLoop *eventLoop, int fd, int mask) {
             if (eventLoop->events[j].mask != AE_NONE) break;
         eventLoop->maxfd = j;
     }
+    /* Must be invoked after the eventLoop mask is modified,
+     * which is required by evport and epoll */
+    aeApiDelEvent(eventLoop, fd, mask);
```

Comment:
Great suggestion!  
We can use the `mask` variable directly to check, since it's the `actual` mask to be deleted.  
Also we can handle the situation that user may only remove `AE_BARRIER` without touching the actual events.  
(Currently there's no `AE_BARRIER` usage and it was always used together with `AE_WRITABLE` in the history, but we can improve robustness by doing so)

A new commit is pushed, please review when you have time ~